tests/unit/findIssueDepthByUrl.test.ts

```diff
@@ -0,0 +1,35 @@
+import { findIssueDepthByUrl } from '../../lib/findIssueDepthByUrl';
+import { testIssueData } from '../fixtures/findIssueDepthByUrl.issueData';
+
+describe('findIssueDepthByUrl', function() {
+
+  it('returns the correct depth for root issue', () => {
+    const actual = findIssueDepthByUrl(testIssueData, 'root');
+    expect(actual).toEqual(0);
+  });
+
+  it('returns the correct depth for child1 issue', () => {
+    const actual = findIssueDepthByUrl(testIssueData, 'child1');
+    expect(actual).toEqual(1);
+  });
+
+  it('returns the correct depth for child2 issue', () => {
+    const actual = findIssueDepthByUrl(testIssueData, 'child2');
+    expect(actual).toEqual(1);
+  });
+
+  it('returns the correct depth for child1-1 issue', () => {
+    const actual = findIssueDepthByUrl(testIssueData, 'child1-1');
+    expect(actual).toEqual(2);
+  });
+
+  it('returns the correct depth for child2-1-1 issue', () => {
+    const actual = findIssueDepthByUrl(testIssueData, 'child2-1-1');
+    expect(actual).toEqual(3);
+  });
+
+  it('returns -1 for urls that arent found', () => {
+    const actual = findIssueDepthByUrl(testIssueData, 'no_test_html_matches_me');
+    expect(actual).toEqual(-1);
+  });
+})
```

Comment:
yea i'm good with that. I try to be careful with fancy test things because it can quickly evolve into hard to maintain tests, but this is a great improvement. thanks =)